From 50ca821068ebfdedd8950690b48eaf2367ede9f2 Mon Sep 17 00:00:00 2001
From: Devajith Valaparambil Sreeramaswamy
 <devajith.valaparambil.sreeramaswamy@cern.ch>
Date: Thu, 31 Oct 2024 09:32:19 +0100
Subject: [PATCH 31/51] Revert "Do not load the source file just to get an
 irrelevant SourceLoc (ROOT-7111)."

This reverts commit b8c917819d88299f1a3762b7bb41f27e52e49ad7.
---
 interpreter/llvm-project/clang/lib/Lex/Lexer.cpp | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/interpreter/llvm-project/clang/lib/Lex/Lexer.cpp b/interpreter/llvm-project/clang/lib/Lex/Lexer.cpp
index 20d0bb7217..50b56265f6 100644
--- a/interpreter/llvm-project/clang/lib/Lex/Lexer.cpp
+++ b/interpreter/llvm-project/clang/lib/Lex/Lexer.cpp
@@ -813,10 +813,6 @@ SourceLocation Lexer::getLocForEndOfToken(SourceLocation Loc, unsigned Offset,
       return {}; // Points inside the macro expansion.
   }
 
-  // Don't hit the file system for ASTReader tokens.
-  if (SM.isLoadedSourceLocation(Loc))
-    return Loc;
-
   unsigned Len = Lexer::MeasureTokenLength(Loc, SM, LangOpts);
   if (Len > Offset)
     Len = Len - Offset;
-- 
2.43.5

